src/pages/DatasetCatalog.js

```diff
@@ -101,6 +102,40 @@ export default function DatasetCatalog(props) {
     init();
   }, [getDatasets]);
 
+  useEffect( () => {
+    const doEnrichment = async() => {
+      spinnerService.showAll();
+      const theCurrentPage = filterToOnlySelected ? currentPageOnlySelected : currentPageAllDatasets;
```

Comment:
There's a complex state machine in this code.  
I believe pulling this into a hook is the right approach, but the immediate impact is the need for managing all of the elements that can change state within the hook, hence why there's some seemingly duplicate code.  If I were to refactor this page, I would get more granular with state because the current approach isn't actually saving enough in certain conditions (for example, searchTable was invoked multiple times).